platform/pulp/platform/models/repository.py

```diff
@@ -1,10 +1,341 @@
+    :type basic_auth_password: models.TextField
+
+    :cvar max_download_bandwidth: The max amount of bandwidth used per download.
+    :type max_download_bandwidth: models.IntegerField
+
+    :cvar max_concurrent_downloads: The number of concurrent downloads permitted.
+    :type max_concurrent_downloads: models.IntegerField
+
+    :cvar download_policy: The policy for downloading content.
+    :type download_policy: models.TextField
+
+    :cvar last_sync: When the last successful synchronization occurred.
+    :type last_sync: models.DateTimeField
+
+    Relations:
+
+    :cvar scratchpad: Arbitrary information stashed by the importer.
+    :type scratchpad: fields.GenericRelation
+    """
+
+    # Download Policies
+    IMMEDIATE = 'immediate'
+    ON_DEMAND = 'on_demand'
+    BACKGROUND = 'background'
+    DOWNLOAD_POLICIES = (
+        (IMMEDIATE, 'Download Immediately'),
+        (ON_DEMAND, 'Download On Demand'),
+        (BACKGROUND, 'Download In Background'))
+
+    feed_url = models.TextField()
+    validate = models.BooleanField(default=False)
+
+    ssl_ca_certificate = models.TextField(blank=True, default='')
+    ssl_client_certificate = models.TextField(blank=True, default='')
+    ssl_client_key = models.TextField(blank=True, default='')
+    ssl_validation = models.BooleanField(default=False)
+
+    proxy_host = models.TextField(blank=True, default='')
+    proxy_port = models.IntegerField(null=True)
+    proxy_user = models.TextField(blank=True, default='')
+    proxy_password = models.TextField(blank=True, default='')
+
+    basic_auth_user = models.TextField(blank=True, default='')
+    basic_auth_password = models.TextField(blank=True, default='')
+
+    max_download_bandwidth = models.IntegerField(null=True)
+    max_concurrent_downloads = models.IntegerField(null=True)
+
+    download_policy = models.TextField(choices=DOWNLOAD_POLICIES)
+    last_sync = models.DateTimeField(blank=True, null=True)
+
+    scratchpad = fields.GenericRelation(Scratchpad)
+
+    class Meta:
+        abstract = True
+        unique_together = ('name', 'repository')
+
+
+class RepositoryImporter(Importer):
+    """
+    A content importer that is associated with a repository..
+
+    Fields:
+
+    Relations:
+
+    :cvar repository: The associated repository.
+    :type repository: models.ForeignKey
+    """
+
+    repository = models.ForeignKey(
+        Repository, related_name='importers', on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ('name', 'repository')
+
+
+class Distributor(Plugin):
+    """
+    An abstract content distributor.
+
+    Fields:
+
+    :cvar relative_path: The (relative) path component of the published url.
+    :type relative_path: models.TextField
+
+    :cvar last_published: When the last successful publish occurred.
+    :type last_published: models.DateTimeField
+
+    Relations:
+
+    """
+    relative_path = models.TextField(blank=True, default='')
```

Comment:
Do we have a plan to ensure there are not collisions? I think this is currently this is done in the repo config conduit.